runtime/main/src/main/java/org/qbicc/runtime/main/VMHelpers.java

```diff
@@ -69,13 +71,14 @@ private static boolean isAssignableTo(Object value, type_id toTypeId, int toDime
         // putchar(':');
         // printInt(toDimensions);
         // putchar('\n');
-        if (toDimensions == 0) {
+        int intDim = toDimensions.intValue();
+        if (intDim == 0) {
             return isAssignableToLeaf(valueTypeId, toTypeId);
         } else if (ObjectModel.is_reference_array(valueTypeId)) {
-            int valueDims = ObjectModel.dimensions_of(value);
+            uint8_t valueDims = ObjectModel.dimensions_of(value);
             if (valueDims == toDimensions) {
                 return isAssignableToLeaf(ObjectModel.element_type_id_of(value), toTypeId);
-            } else if (valueDims > toDimensions) {
+            } else if (valueDims.intValue() > toDimensions.intValue()) {
```

Comment:
not sure this is correct, I could not think of a better way

Reply:
This is correct.  The dimensions values are always between `0` and `256` so the widening casts can be optimized away.